### PR TITLE
feat(DC): CDC Storage support

### PR DESCRIPTION
**Pull request by rahul-kumar-saini**
### Overview
- Previously only Readable and Writable storages were supported
- This doesn't work for the CDC storages
- Introduced `cdc_storage` config type and refactored storage building to support the new type

### CODE REVIEW

File: snuba/datasets/configuration/storage_builder.py [hunk 1]
```diff
@@ -53,27 +57,65 @@ def build_storage_from_config(
...
+        ),
+        QUERY_SPLITTERS: get_query_splitters(
+            config[QUERY_SPLITTERS] if QUERY_SPLITTERS in config else []
+        ),
+        MANDATORY_CONDITION_CHECKERS: get_mandatory_condition_checkers(
+            config[MANDATORY_CONDITION_CHECKERS]
+            if MANDATORY_CONDITION_CHECKERS in config
+            else []
+        )
+        # TODO: Rest of readable storage optional args
```
Comment (volokluev) [inline snuba/datasets/configuration/storage_builder.py:89]: what's left?

File: tests/datasets/configuration/test_storage_loader.py [hunk 2]
```diff
@@ -112,6 +113,15 @@ def _deep_compare_storages(old: Storage, new: Storage) -> None:
             new.get_table_writer().get_stream_loader(),
         )
 
+    if isinstance(old, CdcStorage) or isinstance(new, CdcStorage):
```
Comment (volokluev) [inline tests/datasets/configuration/test_storage_loader.py:116]: Is this code actually running on anything? We don't have any CDC storages in config to compare yet right?

**Review by volokluev — CHANGES_REQUESTED** [review]
This needs tests that define a CDC storage (independent of the ones that exist right now) and makes sure it's parsed and populated correctly

File: tests/datasets/configuration/test_storage_loader.py (hunk 2 above)
Comment (rahul-kumar-saini) [reply tests/datasets/configuration/test_storage_loader.py:116]: This is for #3549, thought I'd add the support in this PR so the other PR is cleaner

File: snuba/datasets/configuration/storage_builder.py (hunk 1 above)
Comment (rahul-kumar-saini) [reply snuba/datasets/configuration/storage_builder.py]: Nothing, good catch

**Review by evanh — APPROVED** [review]

**Pull request merged**